Clarified roles versus permissions usage (#3889)
SoftwareAG/c8y-docs: content/microservice-sdk/general-aspects-bundle/security.md

The following role types are defined for users:

* Required roles: The permissions that are predefined to allow access to {{< product-c8y-iot >}} REST APIs.

> For instance, if a microservice creates measurements using the service user, measurement ADMIN permission must be added as a required role/permission of the application.

Reviewer: ```suggestion
For instance, if a microservice creates measurements using the service user, the measurement ADMIN permission must be added as a required role/permission of the application.
```

> For instance, if a microservice creates measurements using the service user, the measurement ADMIN permission must be added as a required role/permission of the application.

Reviewer 2: #```suggestion
For instance, if a microservice creates measurements using the service user, ADMIN permission for "Measurements" must be added as a required role/permission of the application.
```

> For instance, if a microservice creates measurements using the service user, the measurement ADMIN permission must be added as a required role/permission of the application.

Author: ```suggestion
For instance, if a microservice creates measurements using the service user, the ADMIN permission for "Measurement" must be added as a required permission of the application.
```

Required roles are added to the service users.

> * Roles: The custom roles provided to tenant platform users by the microservice developer.

Author: ```suggestion
* Roles: The custom permissions provided to tenant platform users by the microservice developer.
```

These permissions can be assigned or revoked to the tenant platform users or groups using the Administration application.

> Custom roles/permissions must adhere to this name format in order to be shown in the UI:

Reviewer: actually I think this statement is meant to apply to permissions only - these are just permissions strings not roles

> Custom roles/permissions must adhere to this name format in order to be shown in the UI:

Author: ```suggestion
Custom rpermissions must adhere to this name format to be shown in the UI:
```

> Custom rpermissions must adhere to this name format to be shown in the UI:

Author: ```suggestion
Custom permissions must adhere to this name format to be shown in the UI:
```

<!-- TODO: add/describe a picture of "required roles" and "provided roles" showing a microservice as a block -->

> The roles are set in the [Microservice manifest](#microservice-manifest). For more details about users and roles, refer to [Managing permissions and roles](/standard-tenant/managing-permissions) or to the [User API](https://{{< domain-c8y >}}/api/core/#tag/User-API) in the {{< openapi >}}.

Reviewer: I _think_ this is still referring to permissions not roles

> The roles are set in the [Microservice manifest](#microservice-manifest). For more details about users and roles, refer to [Managing permissions and roles](/standard-tenant/managing-permissions) or to the [User API](https://{{< domain-c8y >}}/api/core/#tag/User-API) in the {{< openapi >}}.

Author: ```suggestion
The permissions are set in the [Microservice manifest](#microservice-manifest). For more details about users, permissions, and roles, refer to [Managing permissions and roles](/standard-tenant/managing-permissions) or to the [User API](https://{{< domain-c8y >}}/api/core/#tag/User-API) in the {{< openapi >}}.
```

{{< c8y-admon-important >}}

> Be aware that the term "role" might be misleading here as the the `roles`field and `requiredRoles` field expect a "permission" string (for example, ROLE_INVENTORY_ADMIN). See also the glossary for the usage of the terms [permission](/glossary/#permission) and [role](/glossary/#role) in the Cumulocity context.

Reviewer: ```suggestion
Be aware that the term "role" might be misleading here as the the `roles` field and `requiredRoles` field expect a "permission" string (for example, ROLE_INVENTORY_ADMIN). See also the glossary for the usage of the terms [permission](/glossary/#permission) and [role](/glossary/#role) in the Cumulocity context.
```

whitespace typo
